common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/LogStore.scala

```diff
@@ -79,10 +80,18 @@ trait LogStore {
    *
    * This method is called when a user requests logs via the API.
    *
-   * @param activation activation to fetch the logs for
+   * @param namespace namespace to fetch the logs for
+   * @param activationId activation to fetch the logs for
+   * @param start activation start
+   * @param end activation end
    * @return the relevant logs
    */
-  def fetchLogs(activation: WhiskActivation, context: UserContext): Future[ActivationLogs]
+  def fetchLogs(namespace: String,
+                activationId: String,
+                start: Instant = Instant.EPOCH,
+                end: Instant = Instant.now(),
+                logs: ActivationLogs,
```

Comment:
This is a little confusing because it looks like "here are the logs, now return me the logs". Maybe add a comment to indicate the logs in the parameter are only not empty if a) the activation record exists and b) the logs are stored embedded in the activation record. It may be better to replace `logs` param with an `Option[WhiskActivation]` parameter, which would only be Some in case the activation record exists. Then, in LogStore impls that require the activation record, they should explicitly fail with a message like "You cannot use LogStore XYZ unless activation storage is enabled" (or have some config to silently ignore these).

Reply:
Yeah good point, it was really confusing for me also at first to realize that in some LogStore implementations the logs are coming from a field inside WhiskActivation, whereas in the external services ones you don't even care about activation.log field. 
I'll add the comments and the Option, but maybe `Option[ActivationLogs]` as we don't have to send the entire object. I will use the Option for `start` and `end` too, as they are not always required also.